tests/test_adjoint.py

```diff
@@ -39,7 +54,13 @@ def test_adjoint_F(self, mkey, shape, kernel, space_order):
         tn = 500.  # Final time
 
         # Create solver from preset
-        solver = acoustic_setup(shape=shape, spacing=[15. for _ in shape], kernel=kernel,
+        if setup_func.__name__ == 'acoustic_setup':
```

Comment:
does it actually need the if/else? Only difference is `kernel` in the kwargs but I "think" tti would just ignore it no?

Reply:
I think the if/else cannot be avoided as the kwargs passed in `tti_setup` are also passed to the `AnisotropicWaveSolver` function and the `ForwardOperator` has a kernel parameter set by default to centered, so the kernel is a positional parameter in `ForwardOperator`

Reply:
I would be passing twice the kernel parameter in ` ForwardOPerator`

Reply:
Sorry, the solution was in my reply ;)